src/tallies/tally_scoring.cpp

```diff
@@ -2473,4 +2474,30 @@ void score_surface_tally(Particle& p, const vector<int>& tallies)
     match.bins_present_ = false;
 }
 
+void score_pht_tally(Particle& p)
+{
+  for (auto i_tally : model::active_tracklength_tallies) {
+    auto& tally {*model::tallies[i_tally]};
+
+    auto filter_iter = FilterBinIter(tally, p);
+    auto score_index = 0;
+
+    double score = p.pht_storage()[simulation::cell_pht];
+
+    for (int i = 0; i < simulation::bins_pht.size() - 1; ++i) {
```

Comment:
Don't do a linear search for the right bin! O(n)? Ain't nobody got time for that. You probably want to use `std::lower_bound` to do this bin search in O(log(n)) time.